src/ff/galois_field.rs

```diff
@@ -47,6 +88,13 @@ macro_rules! bit_array_impl {
                 }
             }
 
+            // TODO (taikiy): Remove this infallible conversion and bring back to TryFrom
+            impl From<u128> for $name {
+                fn from(v: u128) -> Self {
+                    Self::truncate_from(v)
+                }
+            }
```

Comment:
GaloisField (or BitArray) used to implement a fallible conversion of TryFrom. By making the `GaloisField` a subtype of `Field`, we need to exclusively choose either `From` or `TryFrom`. I agree with Martin that we should implement TryFrom to catch any unexpected lossy conversions, but there are literary hundreds of references to `F::from()`, so I'm going to make a separate change to remove `From` from `Field` and replace it by `TryFrom`. For now, `GaloisField` needs to follow the trait bound and implement `From` but this is temporal.